task-manager-backend/src/mail/mongo/mail-mongo.js

```
@@ -0,0 +1,76 @@
+const ApiError = require("../../utils/api-error");
+const Mail = require("./mail-model");
+const {
+    MAIL_CREATE_FAILED,
+    MAIL_FETCH_FAILED,
+    MAIL_UPDATE_FAILED,
+    MAIL_DELETE_FAILED,
+} = require("../../commons/constants");
+
+const createMail = async (data) => {
+    try {
+        const mail = await Mail.create(data);
+        return mail;
+    } catch (error) {
+        throw new ApiError(401, MAIL_CREATE_FAILED);
```

This should be an internal error. As its not related to authentications. Refer this https://blog.hubspot.com/website/api-errors to get a better understand of when we should throw which errors and status codes.
Here as well its not a controller so lets just throw the errors and handle it at controller level

---

Okay, I will update status code in all files with the reference.